src/audio/dai-zephyr.c

```diff
@@ -438,6 +444,47 @@ int dai_common_new(struct dai_data *dd, struct comp_dev *dev,
 	dd->xrun = 0;
 	dd->chan = NULL;
 
+	/* I/O performance init, keep it last so the function does not reach this in case
+	 * of return on error, so that we do not waste a slot
+	 */
+#ifdef CONFIG_SOF_TELEMETRY_IO_PERFORMANCE_MEASUREMENTS
+	enum io_perf_data_item_id perf_type;
+	enum io_perf_data_item_dir perf_dir;
+
+	switch (dai_cfg->type) {
+	case SOF_DAI_INTEL_SSP:
+		perf_type = IO_PERF_I2S_ID;
+		break;
+	case SOF_DAI_INTEL_ALH:
+		perf_type = IO_PERF_SOUND_WIRE_ID;
+		break;
+	case SOF_DAI_INTEL_DMIC:
+		perf_type = IO_PERF_DMIC_ID;
+		break;
+	case SOF_DAI_INTEL_HDA:
+		perf_type = IO_PERF_HDA_ID;
+		break;
+
+	default:
+		perf_type = IO_PERF_INVALID_ID;
+	}
+	if (dai_cfg->direction == SOF_IPC_STREAM_PLAYBACK)
+		perf_dir = IO_PERF_INPUT_DIRECTION;
+	else
+		perf_dir = IO_PERF_OUTPUT_DIRECTION;
+
+	/* ignore perf meas init on case of other dai types */
+	if (perf_type != IO_PERF_INVALID_ID) {
+		struct io_perf_data_item init_data = {perf_type,
+						      cpu_get_id(),
+						      perf_dir,
+						      IO_PERF_POWERED_UP_ENABLED,
+						      IO_PERF_D0IX_POWER_MODE,
+						      0, 0, 0 };
+		io_perf_monitor_init_data(&dd->io_perf_bytes_count, &init_data);
+	}
+#endif
+
 	return 0;
```

Comment:
also in the `INVALID` case?

Reply:
Yes, we should treat performance as something optional. Perf setup failing should not make everything else fail

Reply:
ok, then maybe at least print a warning in the `default` case at line 509 above - not a blocker, can be a follow up patch